python_modules/dagster/dagster/core/definitions/pipeline.py

```diff
@@ -627,7 +632,7 @@ def _get_pipeline_subset_def(
                     deps[_dep_key_of(solid)][input_handle.input_def.name] = DependencyDefinition(
                         solid=output_handle.solid.name, output=output_handle.output_def.name
                     )
-            if graph.dependency_structure.has_dynamic_fan_in_dep(input_handle):
+            elif graph.dependency_structure.has_dynamic_fan_in_dep(input_handle):
```

Comment:
these are equivalent because they're mutually exclusive conditions, right?

Reply:
ya, an input can only have one type of dep